src/subclass/widget.rs

```diff
@@ -353,6 +716,309 @@ where
 //     imp.can_activate_accel(&wrap, signal_id) as glib_sys::gboolean
 // }
 
+unsafe extern "C" fn widget_child_notify<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkWidget,
+    paramptr: *mut gobject_sys::GParamSpec,
+)
+where
+    T: WidgetImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Widget = from_glib_borrow(ptr);
+    let paramwrap: glib::ParamSpec = from_glib_borrow(paramptr);
+
+    imp.child_notify(&wrap, &paramwrap)
+}
+
+unsafe extern "C" fn widget_composited_changed<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkWidget,
+)
+where
+    T: WidgetImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Widget = from_glib_borrow(ptr);
+
+    imp.composited_changed(&wrap)
+}
+
+unsafe extern "C" fn widget_compute_expand<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkWidget,
+    hexpand_ptr: *mut glib_sys::gboolean,
+    vexpand_ptr: *mut glib_sys::gboolean,
+)
+where
+    T: WidgetImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Widget = from_glib_borrow(ptr);
+    let mut hexpand_p: bool = *hexpand_ptr != 0;
+    let mut vexpand_p: bool = *vexpand_ptr != 0;
+
+    imp.compute_expand(&wrap, &mut hexpand_p, &mut vexpand_p)
```

Comment:
Afterwards you need to set the new values of `hexpand_p` / `vexpand_p` to the pointers again

```rust
*hexpand_ptr = hexpand_p.to_glib();
```